src/transfers/MultiAppMooseOkapiTransfer.C

```diff
@@ -169,50 +173,128 @@ MultiAppMooseOkapiTransfer::execute()
           MutableCoefficientsInterface & to_object =
               (this->*getSubAppObject)(_multi_app->appProblemBase(I), _multi_app_object_name, 0);
 
-          // Initialize an array to hold the expansion coefficients we'll receive
-          // from OpenMC. This array holds all of the expansion coefficients
-          // for the cell we specify. For a generic Zernike-Legendre expansion,
-          // the order used by OpenMC is specified in the tallies XML file. It is
-          // assumed that this XML order matches the available slots to write in
-          // the to_aux_scalar variables provided from MOOSE, so this is used to
-          // determine how large of an array to specify here that will be written.
-          // A check is provided in OpenMC to make sure that the length of the
-          // t % coeffs array) matches what is allocated in C++ to receive it.
+          int32_t * filter_indices = nullptr;
+          int32_t num_filter_indices;
+          int err_get =
+              openmc_tally_get_filters(_tally_index, &filter_indices, &num_filter_indices);
+          ErrorHandling::openmc_tally_get_filters(err_get, "MultiAppMooseOkapiTransfer");
+
+          if (num_filter_indices != 2)
+            mooseError(
+                "We expect there to be exactly two filters, one a cell filter and the other an "
```

Comment:
Can you think of any situations where we might want to have more than two filters? For T/H coupling where OpenMC provides power I can't think of any, but I could imagine a scenario where a MultiApp wants to couple in the fast fission power (energy + cell + FE) for a material damage correlation or something. But this could obviously just be added as-needed, so I think this is fine

Reply:
If you want to do a 2D or 3D expansion (e.g., using two Legendres), each would be a separate filter so in that case you would definitely have more than two filters.

Reply:
Oh, right! I was thinking in terms of the older tally scores...

Reply:
Solid points. For multiple expansion filters, we will need to be smart about error checking/loading the results. The MOOSE FET modules expects the order to be x,y,z (given that expansions are defined for each axis). We do have the OpenMC API functions `openmc _*_filter_get_params` where we can retrieve the axis for a filter. This could then inform how to load the results.

@aprilnovak Does any of your immediate work require multi-D expansion filters? At least the "results" test that we had before just used a Zernike expansion (multiplied by a constant from a "Legendre expansion"). I would honestly like to implement this in a follow-on PR but could do it now if you need it

Reply:
I'm mostly working on Pronghorn stuff at the moment, so whatever order of operations works best is fine with me!